library/lua/enable-util.lua

```diff
@@ -0,0 +1,54 @@
+
+local _ENV = mkmodule("enable-util")
+
+local repeatUtil = require('repeat-util')
+
+function assignPersistentGlobal(global_key, global_store, key, value)
```

Comment:
I don't think this is flexible enough in practice. scripts may be modifying fields deep in the state table. The pattern I've usually seen is "make a batch of changes to the state, then persist from the top". However, with that pattern, this function reduces to just the call to `dfhack.persistent.saveSiteData` and the value of abstracting that out into an `enable-util` function becomes questionable